testdata/sqllogictests/explain.slt

```diff
@@ -5,3 +5,28 @@ explain select 1;
 
 statement ok
 explain analyze select 1;
+
+# Test for #1754
+# Ensure `RuntimeGroupExec` is pulled as far up as possible in the `EXPLAIN`ed output
+statement ok
+create temp table t1754 as values (1, 'one'), (2, 'two'), (3, 'three');
+
+query TT
+explain
+select * from t1754
+where column1 != 2
+order by column2
+limit 1;
+----
+logical_plan Limit: skip=0, fetch=1
+  Sort: t1754.column2 ASC NULLS LAST, fetch=1
+    Filter: t1754.column1 != Int64(2)
+      TableScan: t1754 projection=[column1, column2]
+physical_plan RuntimeGroupExec: runtime_preference=local
+  GlobalLimitExec: skip=0, fetch=1
+    SortPreservingMergeExec: [column2@1 ASC NULLS LAST], fetch=1
+      SortExec: fetch=1, expr=[column2@1 ASC NULLS LAST]
+        CoalesceBatchesExec: target_batch_size=8192
+          FilterExec: column1@0 != 2
+            RepartitionExec: partitioning=RoundRobinBatch(10), input_partitions=1
+              MemoryExec: partitions=1, partition_sizes=[2]
```

Comment:
This is a Draft PR because I'm debugging a seemingly faulty test comparison/failure
```sql
  2023-09-19T11:23:43.570162Z ERROR testing::slt::cli: Error while running test `sqllogictests/explain`, error: test fail: query result mismatch:
[SQL] explain
select * from t1754
where column1 != 2
order by column2
limit 1;
[Diff] (-expected|+actual)
    logical_plan Limit: skip=0, fetch=1
      Sort: t1754.column2 ASC NULLS LAST, fetch=1
        Filter: t1754.column1 != Int64(2)
          TableScan: t1754 projection=[column1, column2]
    physical_plan RuntimeGroupExec: runtime_preference=local
      GlobalLimitExec: skip=0, fetch=1
        SortPreservingMergeExec: [column2@1 ASC NULLS LAST], fetch=1
          SortExec: fetch=1, expr=[column2@1 ASC NULLS LAST]
            CoalesceBatchesExec: target_batch_size=8192
              FilterExec: column1@0 != 2
                RepartitionExec: partitioning=RoundRobinBatch(10), input_partitions=1
-                 MemoryExec: partitions=1, partition_sizes=[2]
+                 MemoryExec: partitions=1, partition_sizes=[2]
at /Users/markogrujic/GlareDB/glaredb/testdata/sqllogictests/explain.slt:14

    at crates/testing/src/slt/cli.rs:320 on main ThreadId(1)

Error: Test failures
error: test failed, to rerun pass `-p testing --test sqllogictests`
```
I did try a combination of tabs/spaces/newlines but to no avail.

Reply:
Yeah this is a bit weird, tried some whitespace changes locally too and didn't work.

Also what's the plan for the repartition exec and different number of threads?

```
-               RepartitionExec: partitioning=RoundRobinBatch(10), input_partitions=1
-                 MemoryExec: partitions=1, partition_sizes=[2] 
+               RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1
+                 MemoryExec: partitions=1, partition_sizes=[2]
```

Reply:
For now I omitted this test, hope to get one added in some other PR.

Reply:
@gruuya can we create an issue for the whitespace thing? I'll have to dig a bit deeper into it.

Reply:
Certainly, will open one.

Reply:
Opened #1793 to address this faulty test.